### ISSUE TITLE
Key isn't always found in owned vehicles.

### ISSUE THREAD

**Issue by OhTanoshi** [Xseba360/esx_locksystem#2]
Decided to test this out once i seen you had made a esx version. and it works great when it allows me to find the key. You could probably make it check if player owns the vehicle and if they do make it equal to haskey. i havent quite looked at the code so i dont know the correct names for it but just a idea.

Also thanks for the pull requests on fixing my old garage system.

**Comment by Xseba360** [comment]
There is a MySQL check once the player connects to the server, however there is also an addition that can be made to the vehicle shop script, which automatically adds the keys for the player when a vehicle is bought
https://www.diffchecker.com/beGyQtFh

**Comment by OhTanoshi** [comment]
You would also need to do the same/similar with cars coming out of the garage. only a 50% chance of getting/finding the keys when taking a car out

**Comment by Xseba360** [comment]
@lilfuchs 
https://github.com/Xseba360/esx_locksystem/blob/1f28720979f05470b010884cad07d0d2d7f3d58b/server/server.lua#L18

https://github.com/Xseba360/esx_locksystem/blob/1f28720979f05470b010884cad07d0d2d7f3d58b/server/server.lua#L29

These 2 lines take care of this, when resource included on startup :)

**Comment by OhTanoshi** [comment]
You would have to add lines to garage system also so that when a car is taken out the player gets keys. because if they leave and rejoin its a 50/50 if they will get keys again

**Comment by XxFri3ndlyxX** [comment]
Iv'e been trying to make this work with the latest esx_vechicleshop using the link the provided above  but i rarely get the key upon buying the car. Not sure what to do. The only alternative i had was to make 100% key then disable npc and make a lockpick item to unlock vehicle in the world

**Comment by EGUltraTM** [comment]
Would be nice ix this could be fixed. Have the same problem

**Comment by OhTanoshi** [comment]
Just look at what he suggested with adding to vehicleshops and add it to the garage and it works

**Comment by EGUltraTM** [comment]
Can you maybe help what exactly is meaned there? Then i would do it

**Comment by Xseba360** [comment]
@EGUltraTM https://github.com/Xseba360/esx_locksystem/issues/2#issuecomment-412124815

**Comment by OhTanoshi** [comment]
That check is not exactly reliable. theirs multiple times where i can have a key, store the car take it back out and not be able to find it again, etc. at least with the garage system i was using that's how it was with my eden edit i haven't tested it yet but will soon.

**Comment by EGUltraTM** [comment]
for the vehicle shop this doenst gave me the keys anyway.
Modified the code like above

**Comment by XxFri3ndlyxX** [comment]
Yup same for me like i said. vehicleshop code provided does not work.  :(  I tried to find why but yet im not sure. If i ever find i will let you guys know.

**Comment by amakuu** [comment]
the same. keys isnt always found in vehicles

**Comment by djowroots** [comment]
same for me
